qoffer-infrastructure/src/test/groovy/life/qbic/portal/qoffer2/database/SearchCustomerDataSourceSpec.groovy

```diff
@@ -0,0 +1,69 @@
+package life.qbic.portal.qoffer2.database
+
+import groovy.sql.GroovyRowResult
+import life.qbic.datamodel.dtos.business.AcademicTitleFactory
+import life.qbic.datamodel.dtos.business.Customer
+import life.qbic.portal.portlet.customers.search.SearchCustomerDataSource
+import life.qbic.portal.qoffer2.customers.CustomerDbConnector
+import org.apache.groovy.sql.extensions.SqlExtensions
+import spock.lang.Specification
+
+import java.sql.Connection
+import java.sql.PreparedStatement
+import java.sql.ResultSet
+
+/**
+ * This class tests the implementation of SearchCustomerDataSource
+ *
+ * In this specification every behaviour described by the SearchCustomerDataSource interface should be tested
+ *
+ * @since: 1.0.0
+ */
+class SearchCustomerDataSourceSpec extends Specification{
+    AcademicTitleFactory factory = new AcademicTitleFactory()
+
+    def "CustomerDbConnector shall return matching customers given a first name and a last name"() {
+        given: "a predefined query template"
+        String expectedQuery = "SELECT id, first_name AS firstName, last_name AS lastName, title as academicTitle, email as eMailAddress FROM customer" +
```

Comment:
```suggestion
        String expectedQuery = "SELECT id, first_name AS firstName, last_name AS lastName, title as academicTitle, email as eMailAddress FROM person" +
```
I think we agreed that we don't have an explicit customer table anymore

Reply:
Should be changed in a separate PR addressing the problem. I guess